File: C++/Player.cpp
#include "Player.h"
#include <string>
#include <sstream>

using namespace std;
Player::Player()
{

}

Player::Player(string ipos, string iname, int isalary, int ippg)
{
	name = iname;
	pos = ipos;
	salary = isalary;
	ppg = ippg;
}


Player::~Player()
{
}


string Player::position()
{
	return pos;
}


string Player::print()
{
	ostringstream oss;
	oss << name + " " + pos + " ";
	oss << salary;
	oss << " ";
	oss << ppg;
	return oss.str();
}


string Player::pname()
{
	return name;
}

File: Optimizer/src/code/Player.java
package code;

public class Player {

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((name == null) ? 0 : name.hashCode());
		result = prime * result + ((pos == null) ? 0 : pos.hashCode());
		long temp;
		temp = Double.doubleToLongBits(ppg);
		result = prime * result + (int) (temp ^ (temp >>> 32));
		temp = Double.doubleToLongBits(sal);
		result = prime * result + (int) (temp ^ (temp >>> 32));
		temp = Double.doubleToLongBits(value);
		result = prime * result + (int) (temp ^ (temp >>> 32));
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (!(obj instanceof Player))
			return false;
		Player other = (Player) obj;
		if (name == null) {
			if (other.name != null)
				return false;
		} else if (!name.equals(other.name))
			return false;
		if (pos == null) {
			if (other.pos != null)
				return false;
		} else if (!pos.equals(other.pos))
			return false;
		if (Double.doubleToLongBits(ppg) != Double.doubleToLongBits(other.ppg))
			return false;
		if (Double.doubleToLongBits(sal) != Double.doubleToLongBits(other.sal))
			return false;
		if (Double.doubleToLongBits(value) != Double.doubleToLongBits(other.value))
			return false;
		return true;
	}

	public String name;
	public String pos;
	public double ppg;
	public double sal;
	public double value;
	
	public Player(String iname, String ipos, double ippg, double isal){
		name = iname;
		pos = ipos;
		ppg = ippg;
		sal = isal;
		value = ippg/isal;			// Working idea for the algo
	}
}
